proposed/runtime-binding.md

```diff
@@ -0,0 +1,265 @@
+
+We will offer a configuration knobs for hosts that enable artificial roll-forward. These knobs play a similar role to [LockClrVersion](https://docs.microsoft.com/dotnet/framework/unmanaged-api/hosting/lockclrversion-function).
+
+## Configuration Knobs
+
+The host exposes the following configuration knobs that you can use to control binding behavior:
+
+### DOTNET_ROLL_FORWARD_ON_NO_CANDIDATE_FX
+
+This setting is described in [Roll Forward On No Candidate Fx](https://github.com/dotnet/core-setup/blob/master/Documentation/design-docs/roll-forward-on-no-candidate-fx.md). It does not have the right behavior or UX. We will deprecate this setting for 3.0+.
+
+### RollForward
+
+We will introduce a new knob, called `RollForward`, with the following values:
+
+* `patch` -- Roll forward to the highest patch verison. This disables minor version roll forward.
+* `minor` -- Roll forward to the lowest higher minor verison, if requested minor version is missing.
+* `major` -- Roll forward to lowest higher major version, and lowest minor version, if requested major version is missing
+* `UseLatestMinor` -- Roll forward to latest minor version, even if requested minor version is on machine. Intended for COM hosting scenario.
+* `UseLatestMajor` -- Roll forward to latest major and highest minor version, even if requested major is on machine. Intended for COM hosting scenario.
+
+There is no default setting. See **Configuration Precedence** for more information.
+
+Can be specified via the following ways:
+
+* Project file property: `RollForward`
+* Runtime configuration file property: `rollForward`
+* Environment variable: `DOTNET_ROLL_FORWARD`
+* Command line argument: `--roll-forward`
+
+### fx-version
+
+You can specify the exact desired runtime version using the command line argument '--fx-version'. In this case, only the specified version will be accepted, even if patch roll forward is enabled. The expected behavior would be the same in the example above.
+
+The following example shows how to use this knob:
+
+```console
+C:\testapps\twotwoapp>dotnet --fx-version 3.0.0 bin\Debug\netcoreapp2.2\twotwoapp.dll
+```
+
+The example above will attempt to load the 3.0.0 runtime version. If it isn't found, the host will print an error message.
+
+Another example:
+
+```console
+C:\testapps\threezeroapp>bin\Debug\netcoreapp3.0\threezero.exe --fx-version 3.0.0 --roll-forward patch
+```
+
+The example above will attempt to load the highest `3.0` patch version, which may be `3.0.0` or `3.0.27`, for example, whichever is the highest path version on the machine.
+
+### Configuration Precedence
+
+The host will consult these settings in order (later scopes take precendence over earlier ones):
+
+* `.runtimeconfig.json` properties (AKA "json")
+* CLI arguments
+* Environment variables (AKA "ENV")
+
```

Comment:
Earlier a project (.csproj/.vbproj) setting was proposed. It's not mentioned here in precedence. Is that because it rolls into the runtimeconfig? Cool, but we should mention it I think.

Reply:
>  Is that because it rolls into the runtimeconfig? 

Yes. Project values have no meaning at runtime. I will add a note for that.